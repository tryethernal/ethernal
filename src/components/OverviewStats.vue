<template>
    <v-card class="bg-surface" height="100%">
        <v-card-text>
            <v-row class="stats-grid">
                <v-col cols="12" sm="6" class="stat-item d-flex flex-column">
                    <StatNumber
                        title="Latest Block"
                        :value="currentWorkspaceStore.currentBlock.number !== null && currentWorkspaceStore.currentBlock.number !== undefined ? currentWorkspaceStore.currentBlock.number.toLocaleString() : ''"
                        :raw="true"
                        :loading="currentWorkspaceStore.currentBlock.number === undefined || currentWorkspaceStore.currentBlock.number === null"
                        icon="mdi-cube-outline"
                        :border="false"
                    />
                    <v-divider class="mt-5" width="90%"></v-divider>
                </v-col>
                <v-col cols="12" sm="6" class="stat-item d-flex flex-column">
                    <StatNumber
                        title="Total Tx Count"
                        :value="txCountTotal"
                        :loading="txCountTotalLoading"
                        :border="false"
                    />
                    <v-divider class="mt-5" width="90%"></v-divider>
                </v-col>
                <v-col cols="12" sm="6" class="stat-item d-flex flex-column">
                    <StatNumber
                        title="24h Tx Count"
                        :value="txCount24h"
                        :loading="txCount24hLoading"
                        :border="false"
                    />
                    <v-divider class="mt-5" width="90%"></v-divider>
                </v-col>
                <v-col cols="12" sm="6" class="stat-item d-flex flex-column">
                    <StatNumber
                        title="Total Active Wallets Count"
                        infoTooltip="An active wallet is an address that has sent at least one transaction."
                        :value="activeWalletCount"
                        :loading="activeWalletCountLoading"
                        :border="false"
                    />
                    <v-divider class="mt-5" width="90%"></v-divider>
                </v-col>
                <v-col cols="12" sm="6" class="stat-item d-flex flex-column" v-if="explorerStore.totalSupply">
                    <StatNumber
                        title="Total Supply"
                        :raw="true"
                        :value="formattedTotalSupply"
                        :loading="!formattedTotalSupply"
                        :border="false"
                    />
                </v-col>
            </v-row>
        </v-card-text>
    </v-card>
</template>

<script setup>
import { ref, onMounted, inject, computed } from 'vue';
import StatNumber from './StatNumber.vue';

import { useCurrentWorkspaceStore } from '../stores/currentWorkspace';
import { useExplorerStore } from '../stores/explorer';

const currentWorkspaceStore = useCurrentWorkspaceStore();
const explorerStore = useExplorerStore();

const $server = inject('$server');
const $fromWei = inject('$fromWei');

// Reactive state
const activeWalletCount = ref(0);
const activeWalletCountLoading = ref(false);
const txCountTotal = ref(0);
const txCountTotalLoading = ref(false);
const txCount24h = ref(0);
const txCount24hLoading = ref(false);

// Data fetching methods
const getActiveWalletCount = async () => {
    activeWalletCountLoading.value = true;
    try {
        const { data: { count }} = await $server.getActiveWalletCount();
        activeWalletCount.value = count;
    } catch (error) {
        console.error('Error fetching active wallet count:', error);
    } finally {
        activeWalletCountLoading.value = false;
    }
};

const getTxCountTotal = async () => {
    txCountTotalLoading.value = true;
    try {
        const { data: { count }} = await $server.getTxCountTotal();
        txCountTotal.value = count;
    } catch (error) {
        console.error('Error fetching total transaction count:', error);
    } finally {
        txCountTotalLoading.value = false;
    }
};

const getTxCount24h = async () => {
    txCount24hLoading.value = true;
    try {
        const { data: { count }} = await $server.getTxCount24h();
        txCount24h.value = count;
    } catch (error) {
        console.error('Error fetching 24h transaction count:', error);
    } finally {
        txCount24hLoading.value = false;
    }
};

const formattedTotalSupply = computed(() => {
    return $fromWei(explorerStore.totalSupply, 'ether', explorerStore.token);
});

// Initialize data
onMounted(() => {
    getActiveWalletCount();
    getTxCountTotal();
    getTxCount24h();
});
</script>

<style scoped>
.stats-grid {
    padding: 1rem;
}

.stat-item {
    display: flex;
    align-items: stretch;
}

.stat-item :deep(.v-card) {
    width: 100%;
}

/* Hide dividers for last row */
@media (min-width: 600px) {
    /* On 2-column layout, hide last 2 items' dividers */
    .stat-item:nth-last-child(1) .v-divider,
    .stat-item:nth-last-child(2) .v-divider {
        display: none;
    }
}

@media (max-width: 599px) {
    /* On mobile, only hide last item's divider */
    .stat-item:last-child .v-divider {
        display: none;
    }
}

.bg-surface {
    background-color: var(--card-background);
    box-shadow: 0 4px 6px rgba(0,0,0,0.1) !important;
}
</style>
