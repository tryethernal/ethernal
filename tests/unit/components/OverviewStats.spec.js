import OverviewStats from '@/components/OverviewStats.vue';

const stubs = ['StatNumber'];

describe('OverviewStats.vue', () => {
    beforeEach(() => {
        vi.spyOn(server, 'getActiveWalletCount').mockResolvedValueOnce({ data: { count: 100 } });
        vi.spyOn(server, 'getTxCountTotal').mockResolvedValueOnce({ data: { count: 1000 } });
        vi.spyOn(server, 'getTxCount24h').mockResolvedValueOnce({ data: { count: 50 } });
    });

    it('Should show the component with all stats', async () => {
        const fromWei = vi.fn().mockReturnValue('1.0 ETH');
        const wrapper = mount(OverviewStats, {
            global: {
                stubs,
                provide: {
                    $fromWei: fromWei
                },
                plugins: [createTestingPinia({
                    initialState: {
                        currentWorkspace: {
                            currentBlock: {
                                number: 12345
                            }
                        },
                        explorer: {
                            totalSupply: '1000000000000000000',
                            token: {
                                symbol: 'ETH',
                                decimals: 18
                            }
                        }
                    }
                })]
            }
        });
        await flushPromises();

        expect(wrapper.html()).toMatchSnapshot();
        expect(fromWei).toHaveBeenCalledWith('1000000000000000000', 'ether', { symbol: 'ETH', decimals: 18 });
    });

    it('Should show loading state while fetching data', async () => {
        const fromWei = vi.fn().mockReturnValue('0.0 ETH');
        const wrapper = mount(OverviewStats, {
            global: {
                stubs,
                provide: {
                    $fromWei: fromWei
                },
                plugins: [createTestingPinia({
                    initialState: {
                        currentWorkspace: {
                            currentBlock: {
                                number: null
                            }
                        },
                        explorer: {
                            totalSupply: '0',
                            token: {
                                symbol: 'ETH',
                                decimals: 18
                            }
                        }
                    }
                })]
            }
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('Should handle server errors gracefully', async () => {
        vi.spyOn(server, 'getActiveWalletCount').mockRejectedValueOnce(new Error('Server error'));
        vi.spyOn(server, 'getTxCountTotal').mockRejectedValueOnce(new Error('Server error'));
        vi.spyOn(server, 'getTxCount24h').mockRejectedValueOnce(new Error('Server error'));

        const fromWei = vi.fn().mockReturnValue('1.0 ETH');
        const wrapper = mount(OverviewStats, {
            global: {
                stubs,
                provide: {
                    $fromWei: fromWei
                },
                plugins: [createTestingPinia({
                    initialState: {
                        currentWorkspace: {
                            currentBlock: {
                                number: 12345
                            }
                        },
                        explorer: {
                            totalSupply: '1000000000000000000',
                            token: {
                                symbol: 'ETH',
                                decimals: 18
                            }
                        }
                    }
                })]
            }
        });
        await flushPromises();

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('Should not show total supply when not available', async () => {
        const fromWei = vi.fn().mockReturnValue('0.0 ETH');
        const wrapper = mount(OverviewStats, {
            global: {
                stubs,
                provide: {
                    $fromWei: fromWei
                },
                plugins: [createTestingPinia({
                    initialState: {
                        currentWorkspace: {
                            currentBlock: {
                                number: 12345
                            }
                        },
                        explorer: {
                            totalSupply: null,
                            token: {
                                symbol: 'ETH',
                                decimals: 18
                            }
                        }
                    }
                })]
            }
        });
        await flushPromises();

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('Should hide dividers correctly on different screen sizes', async () => {
        const fromWei = vi.fn().mockReturnValue('1.0 ETH');
        const wrapper = mount(OverviewStats, {
            global: {
                stubs,
                provide: {
                    $fromWei: fromWei
                },
                plugins: [createTestingPinia({
                    initialState: {
                        currentWorkspace: {
                            currentBlock: {
                                number: 12345
                            }
                        },
                        explorer: {
                            totalSupply: '1000000000000000000',
                            token: {
                                symbol: 'ETH',
                                decimals: 18
                            }
                        }
                    }
                })]
            }
        });
        await flushPromises();

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('Should show block number when it is 0', async () => {
        const fromWei = vi.fn().mockReturnValue('1.0 ETH');
        const wrapper = mount(OverviewStats, {
            global: {
                stubs,
                provide: {
                    $fromWei: fromWei
                },
                plugins: [createTestingPinia({
                    initialState: {
                        currentWorkspace: {
                            currentBlock: {
                                number: 0
                            }
                        },
                        explorer: {
                            totalSupply: '1000000000000000000',
                            token: {
                                symbol: 'ETH',
                                decimals: 18
                            }
                        }
                    }
                })]
            }
        });
        await flushPromises();

        expect(wrapper.html()).toMatchSnapshot();
    });
}); 