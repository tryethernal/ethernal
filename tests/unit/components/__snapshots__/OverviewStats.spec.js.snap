// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OverviewStats.vue > Should handle server errors gracefully 1`] = `
"<div data-v-1ebb31f3="" class="v-card v-theme--light v-card--density-default v-card--variant-elevated bg-surface" style="height: 100%;">
  <!---->
  <div class="v-card__loader">
    <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
      <!---->
      <div class="v-progress-linear__background"></div>
      <div class="v-progress-linear__buffer" style="width: 0%;"></div>
      <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
        <div class="v-progress-linear__indeterminate">
          <div class="v-progress-linear__indeterminate long"></div>
          <div class="v-progress-linear__indeterminate short"></div>
        </div>
      </transition-stub>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  <div data-v-1ebb31f3="" class="v-card-text">
    <div data-v-1ebb31f3="" class="v-row stats-grid">
      <div data-v-1ebb31f3="" class="v-col-sm-6 v-col-12 stat-item d-flex flex-column">
        <stat-number-stub data-v-1ebb31f3="" title="Latest Block" loading="false" long="false" raw="true" border="false" icon="mdi-cube-outline" value="12,345"></stat-number-stub>
        <hr data-v-1ebb31f3="" class="v-divider v-theme--light mt-5" aria-orientation="horizontal" role="separator" width="90%">
      </div>
      <div data-v-1ebb31f3="" class="v-col-sm-6 v-col-12 stat-item d-flex flex-column">
        <stat-number-stub data-v-1ebb31f3="" title="Total Tx Count" loading="false" long="false" raw="false" border="false" value="0"></stat-number-stub>
        <hr data-v-1ebb31f3="" class="v-divider v-theme--light mt-5" aria-orientation="horizontal" role="separator" width="90%">
      </div>
      <div data-v-1ebb31f3="" class="v-col-sm-6 v-col-12 stat-item d-flex flex-column">
        <stat-number-stub data-v-1ebb31f3="" title="24h Tx Count" loading="false" long="false" raw="false" border="false" value="0"></stat-number-stub>
        <hr data-v-1ebb31f3="" class="v-divider v-theme--light mt-5" aria-orientation="horizontal" role="separator" width="90%">
      </div>
      <div data-v-1ebb31f3="" class="v-col-sm-6 v-col-12 stat-item d-flex flex-column">
        <stat-number-stub data-v-1ebb31f3="" title="Total Active Wallets Count" infotooltip="An active wallet is an address that has sent at least one transaction." loading="false" long="false" raw="false" border="false" value="0"></stat-number-stub>
        <hr data-v-1ebb31f3="" class="v-divider v-theme--light mt-5" aria-orientation="horizontal" role="separator" width="90%">
      </div>
      <div data-v-1ebb31f3="" class="v-col-sm-6 v-col-12 stat-item d-flex flex-column">
        <stat-number-stub data-v-1ebb31f3="" title="Total Supply" loading="false" long="false" raw="true" border="false" value="1.0 ETH"></stat-number-stub>
      </div>
    </div>
  </div>
  <!---->
  <!----><span class="v-card__underlay"></span>
</div>"
`;

exports[`OverviewStats.vue > Should hide dividers correctly on different screen sizes 1`] = `
"<div data-v-1ebb31f3="" class="v-card v-theme--light v-card--density-default v-card--variant-elevated bg-surface" style="height: 100%;">
  <!---->
  <div class="v-card__loader">
    <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
      <!---->
      <div class="v-progress-linear__background"></div>
      <div class="v-progress-linear__buffer" style="width: 0%;"></div>
      <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
        <div class="v-progress-linear__indeterminate">
          <div class="v-progress-linear__indeterminate long"></div>
          <div class="v-progress-linear__indeterminate short"></div>
        </div>
      </transition-stub>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  <div data-v-1ebb31f3="" class="v-card-text">
    <div data-v-1ebb31f3="" class="v-row stats-grid">
      <div data-v-1ebb31f3="" class="v-col-sm-6 v-col-12 stat-item d-flex flex-column">
        <stat-number-stub data-v-1ebb31f3="" title="Latest Block" loading="false" long="false" raw="true" border="false" icon="mdi-cube-outline" value="12,345"></stat-number-stub>
        <hr data-v-1ebb31f3="" class="v-divider v-theme--light mt-5" aria-orientation="horizontal" role="separator" width="90%">
      </div>
      <div data-v-1ebb31f3="" class="v-col-sm-6 v-col-12 stat-item d-flex flex-column">
        <stat-number-stub data-v-1ebb31f3="" title="Total Tx Count" loading="false" long="false" raw="false" border="false" value="1000"></stat-number-stub>
        <hr data-v-1ebb31f3="" class="v-divider v-theme--light mt-5" aria-orientation="horizontal" role="separator" width="90%">
      </div>
      <div data-v-1ebb31f3="" class="v-col-sm-6 v-col-12 stat-item d-flex flex-column">
        <stat-number-stub data-v-1ebb31f3="" title="24h Tx Count" loading="false" long="false" raw="false" border="false" value="50"></stat-number-stub>
        <hr data-v-1ebb31f3="" class="v-divider v-theme--light mt-5" aria-orientation="horizontal" role="separator" width="90%">
      </div>
      <div data-v-1ebb31f3="" class="v-col-sm-6 v-col-12 stat-item d-flex flex-column">
        <stat-number-stub data-v-1ebb31f3="" title="Total Active Wallets Count" infotooltip="An active wallet is an address that has sent at least one transaction." loading="false" long="false" raw="false" border="false" value="100"></stat-number-stub>
        <hr data-v-1ebb31f3="" class="v-divider v-theme--light mt-5" aria-orientation="horizontal" role="separator" width="90%">
      </div>
      <div data-v-1ebb31f3="" class="v-col-sm-6 v-col-12 stat-item d-flex flex-column">
        <stat-number-stub data-v-1ebb31f3="" title="Total Supply" loading="false" long="false" raw="true" border="false" value="1.0 ETH"></stat-number-stub>
      </div>
    </div>
  </div>
  <!---->
  <!----><span class="v-card__underlay"></span>
</div>"
`;

exports[`OverviewStats.vue > Should not show total supply when not available 1`] = `
"<div data-v-1ebb31f3="" class="v-card v-theme--light v-card--density-default v-card--variant-elevated bg-surface" style="height: 100%;">
  <!---->
  <div class="v-card__loader">
    <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
      <!---->
      <div class="v-progress-linear__background"></div>
      <div class="v-progress-linear__buffer" style="width: 0%;"></div>
      <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
        <div class="v-progress-linear__indeterminate">
          <div class="v-progress-linear__indeterminate long"></div>
          <div class="v-progress-linear__indeterminate short"></div>
        </div>
      </transition-stub>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  <div data-v-1ebb31f3="" class="v-card-text">
    <div data-v-1ebb31f3="" class="v-row stats-grid">
      <div data-v-1ebb31f3="" class="v-col-sm-6 v-col-12 stat-item d-flex flex-column">
        <stat-number-stub data-v-1ebb31f3="" title="Latest Block" loading="false" long="false" raw="true" border="false" icon="mdi-cube-outline" value="12,345"></stat-number-stub>
        <hr data-v-1ebb31f3="" class="v-divider v-theme--light mt-5" aria-orientation="horizontal" role="separator" width="90%">
      </div>
      <div data-v-1ebb31f3="" class="v-col-sm-6 v-col-12 stat-item d-flex flex-column">
        <stat-number-stub data-v-1ebb31f3="" title="Total Tx Count" loading="false" long="false" raw="false" border="false" value="1000"></stat-number-stub>
        <hr data-v-1ebb31f3="" class="v-divider v-theme--light mt-5" aria-orientation="horizontal" role="separator" width="90%">
      </div>
      <div data-v-1ebb31f3="" class="v-col-sm-6 v-col-12 stat-item d-flex flex-column">
        <stat-number-stub data-v-1ebb31f3="" title="24h Tx Count" loading="false" long="false" raw="false" border="false" value="50"></stat-number-stub>
        <hr data-v-1ebb31f3="" class="v-divider v-theme--light mt-5" aria-orientation="horizontal" role="separator" width="90%">
      </div>
      <div data-v-1ebb31f3="" class="v-col-sm-6 v-col-12 stat-item d-flex flex-column">
        <stat-number-stub data-v-1ebb31f3="" title="Total Active Wallets Count" infotooltip="An active wallet is an address that has sent at least one transaction." loading="false" long="false" raw="false" border="false" value="100"></stat-number-stub>
        <hr data-v-1ebb31f3="" class="v-divider v-theme--light mt-5" aria-orientation="horizontal" role="separator" width="90%">
      </div>
      <!--v-if-->
    </div>
  </div>
  <!---->
  <!----><span class="v-card__underlay"></span>
</div>"
`;

exports[`OverviewStats.vue > Should show block number when it is 0 1`] = `
"<div data-v-1ebb31f3="" class="v-card v-theme--light v-card--density-default v-card--variant-elevated bg-surface" style="height: 100%;">
  <!---->
  <div class="v-card__loader">
    <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
      <!---->
      <div class="v-progress-linear__background"></div>
      <div class="v-progress-linear__buffer" style="width: 0%;"></div>
      <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
        <div class="v-progress-linear__indeterminate">
          <div class="v-progress-linear__indeterminate long"></div>
          <div class="v-progress-linear__indeterminate short"></div>
        </div>
      </transition-stub>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  <div data-v-1ebb31f3="" class="v-card-text">
    <div data-v-1ebb31f3="" class="v-row stats-grid">
      <div data-v-1ebb31f3="" class="v-col-sm-6 v-col-12 stat-item d-flex flex-column">
        <stat-number-stub data-v-1ebb31f3="" title="Latest Block" loading="false" long="false" raw="true" border="false" icon="mdi-cube-outline" value="0"></stat-number-stub>
        <hr data-v-1ebb31f3="" class="v-divider v-theme--light mt-5" aria-orientation="horizontal" role="separator" width="90%">
      </div>
      <div data-v-1ebb31f3="" class="v-col-sm-6 v-col-12 stat-item d-flex flex-column">
        <stat-number-stub data-v-1ebb31f3="" title="Total Tx Count" loading="false" long="false" raw="false" border="false" value="1000"></stat-number-stub>
        <hr data-v-1ebb31f3="" class="v-divider v-theme--light mt-5" aria-orientation="horizontal" role="separator" width="90%">
      </div>
      <div data-v-1ebb31f3="" class="v-col-sm-6 v-col-12 stat-item d-flex flex-column">
        <stat-number-stub data-v-1ebb31f3="" title="24h Tx Count" loading="false" long="false" raw="false" border="false" value="50"></stat-number-stub>
        <hr data-v-1ebb31f3="" class="v-divider v-theme--light mt-5" aria-orientation="horizontal" role="separator" width="90%">
      </div>
      <div data-v-1ebb31f3="" class="v-col-sm-6 v-col-12 stat-item d-flex flex-column">
        <stat-number-stub data-v-1ebb31f3="" title="Total Active Wallets Count" infotooltip="An active wallet is an address that has sent at least one transaction." loading="false" long="false" raw="false" border="false" value="100"></stat-number-stub>
        <hr data-v-1ebb31f3="" class="v-divider v-theme--light mt-5" aria-orientation="horizontal" role="separator" width="90%">
      </div>
      <div data-v-1ebb31f3="" class="v-col-sm-6 v-col-12 stat-item d-flex flex-column">
        <stat-number-stub data-v-1ebb31f3="" title="Total Supply" loading="false" long="false" raw="true" border="false" value="1.0 ETH"></stat-number-stub>
      </div>
    </div>
  </div>
  <!---->
  <!----><span class="v-card__underlay"></span>
</div>"
`;

exports[`OverviewStats.vue > Should show loading state while fetching data 1`] = `
"<div data-v-1ebb31f3="" class="v-card v-theme--light v-card--density-default v-card--variant-elevated bg-surface" style="height: 100%;">
  <!---->
  <div class="v-card__loader">
    <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
      <!---->
      <div class="v-progress-linear__background"></div>
      <div class="v-progress-linear__buffer" style="width: 0%;"></div>
      <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
        <div class="v-progress-linear__indeterminate">
          <div class="v-progress-linear__indeterminate long"></div>
          <div class="v-progress-linear__indeterminate short"></div>
        </div>
      </transition-stub>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  <div data-v-1ebb31f3="" class="v-card-text">
    <div data-v-1ebb31f3="" class="v-row stats-grid">
      <div data-v-1ebb31f3="" class="v-col-sm-6 v-col-12 stat-item d-flex flex-column">
        <stat-number-stub data-v-1ebb31f3="" title="Latest Block" loading="true" long="false" raw="true" border="false" icon="mdi-cube-outline" value=""></stat-number-stub>
        <hr data-v-1ebb31f3="" class="v-divider v-theme--light mt-5" aria-orientation="horizontal" role="separator" width="90%">
      </div>
      <div data-v-1ebb31f3="" class="v-col-sm-6 v-col-12 stat-item d-flex flex-column">
        <stat-number-stub data-v-1ebb31f3="" title="Total Tx Count" loading="false" long="false" raw="false" border="false" value="0"></stat-number-stub>
        <hr data-v-1ebb31f3="" class="v-divider v-theme--light mt-5" aria-orientation="horizontal" role="separator" width="90%">
      </div>
      <div data-v-1ebb31f3="" class="v-col-sm-6 v-col-12 stat-item d-flex flex-column">
        <stat-number-stub data-v-1ebb31f3="" title="24h Tx Count" loading="false" long="false" raw="false" border="false" value="0"></stat-number-stub>
        <hr data-v-1ebb31f3="" class="v-divider v-theme--light mt-5" aria-orientation="horizontal" role="separator" width="90%">
      </div>
      <div data-v-1ebb31f3="" class="v-col-sm-6 v-col-12 stat-item d-flex flex-column">
        <stat-number-stub data-v-1ebb31f3="" title="Total Active Wallets Count" infotooltip="An active wallet is an address that has sent at least one transaction." loading="false" long="false" raw="false" border="false" value="0"></stat-number-stub>
        <hr data-v-1ebb31f3="" class="v-divider v-theme--light mt-5" aria-orientation="horizontal" role="separator" width="90%">
      </div>
      <div data-v-1ebb31f3="" class="v-col-sm-6 v-col-12 stat-item d-flex flex-column">
        <stat-number-stub data-v-1ebb31f3="" title="Total Supply" loading="false" long="false" raw="true" border="false" value="0.0 ETH"></stat-number-stub>
      </div>
    </div>
  </div>
  <!---->
  <!----><span class="v-card__underlay"></span>
</div>"
`;

exports[`OverviewStats.vue > Should show the component with all stats 1`] = `
"<div data-v-1ebb31f3="" class="v-card v-theme--light v-card--density-default v-card--variant-elevated bg-surface" style="height: 100%;">
  <!---->
  <div class="v-card__loader">
    <div class="v-progress-linear v-theme--light v-locale--is-ltr" style="top: 0px; height: 0px; --v-progress-linear-height: 2px;" role="progressbar" aria-hidden="true" aria-valuemin="0" aria-valuemax="100">
      <!---->
      <div class="v-progress-linear__background"></div>
      <div class="v-progress-linear__buffer" style="width: 0%;"></div>
      <transition-stub name="fade-transition" appear="false" persisted="false" css="true">
        <div class="v-progress-linear__indeterminate">
          <div class="v-progress-linear__indeterminate long"></div>
          <div class="v-progress-linear__indeterminate short"></div>
        </div>
      </transition-stub>
      <!---->
    </div>
  </div>
  <!---->
  <!---->
  <div data-v-1ebb31f3="" class="v-card-text">
    <div data-v-1ebb31f3="" class="v-row stats-grid">
      <div data-v-1ebb31f3="" class="v-col-sm-6 v-col-12 stat-item d-flex flex-column">
        <stat-number-stub data-v-1ebb31f3="" title="Latest Block" loading="false" long="false" raw="true" border="false" icon="mdi-cube-outline" value="12,345"></stat-number-stub>
        <hr data-v-1ebb31f3="" class="v-divider v-theme--light mt-5" aria-orientation="horizontal" role="separator" width="90%">
      </div>
      <div data-v-1ebb31f3="" class="v-col-sm-6 v-col-12 stat-item d-flex flex-column">
        <stat-number-stub data-v-1ebb31f3="" title="Total Tx Count" loading="false" long="false" raw="false" border="false" value="1000"></stat-number-stub>
        <hr data-v-1ebb31f3="" class="v-divider v-theme--light mt-5" aria-orientation="horizontal" role="separator" width="90%">
      </div>
      <div data-v-1ebb31f3="" class="v-col-sm-6 v-col-12 stat-item d-flex flex-column">
        <stat-number-stub data-v-1ebb31f3="" title="24h Tx Count" loading="false" long="false" raw="false" border="false" value="50"></stat-number-stub>
        <hr data-v-1ebb31f3="" class="v-divider v-theme--light mt-5" aria-orientation="horizontal" role="separator" width="90%">
      </div>
      <div data-v-1ebb31f3="" class="v-col-sm-6 v-col-12 stat-item d-flex flex-column">
        <stat-number-stub data-v-1ebb31f3="" title="Total Active Wallets Count" infotooltip="An active wallet is an address that has sent at least one transaction." loading="false" long="false" raw="false" border="false" value="100"></stat-number-stub>
        <hr data-v-1ebb31f3="" class="v-divider v-theme--light mt-5" aria-orientation="horizontal" role="separator" width="90%">
      </div>
      <div data-v-1ebb31f3="" class="v-col-sm-6 v-col-12 stat-item d-flex flex-column">
        <stat-number-stub data-v-1ebb31f3="" title="Total Supply" loading="false" long="false" raw="true" border="false" value="1.0 ETH"></stat-number-stub>
      </div>
    </div>
  </div>
  <!---->
  <!----><span class="v-card__underlay"></span>
</div>"
`;
